### PR TITLE
Workspace Admin Apis And Minor Fixes

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: apps/bamboohr/migrations/0003_auto_20221212_1052.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('bamboohr', '0002_configuration'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
```
Comment (NileshPant1999) [inline apps/bamboohr/migrations/0003_auto_20221212_1052.py:13]: I have added this manually, was getting an error while creating the migrations found this on stackoverflow

File: apps/orgs/views.py [hunk 2]
```diff
@@ -103,6 +111,13 @@ def update(self, request, *args, **kwargs):
                 'Error while creating Workato Workspace org_id - %s in Fyle %s',
                 org.id, exception.message
             )
+
+            if 'message' in exception.message and 'external id has already been taken' in exception.message['message'].lower():
```
Comment (ashwin1111) [inline apps/orgs/views.py:115]: `external has already been taken`

File: apps/bamboohr/signals.py [hunk 3]
```diff
@@ -7,9 +7,10 @@
...
 
-@receiver(post_save, sender=Configuration)
-def run_post_configuration_triggers(sender, instance: Configuration, **kwargs):
+@receiver(pre_save, sender=Configuration)
+def run_pre_save_configuration_triggers(sender, instance: Configuration, **kwargs):
     connector = Workato()
-    org = Org.objects.get(id=instance.org_id)
     if instance.recipe_status:
+        org = Org.objects.get(id=instance.org_id)
+        print('nilesh')
```
Comment (Shwetabhk) [inline apps/bamboohr/signals.py:15]: remove print

File: apps/bamboohr/signals.py [hunk 4]
```diff
@@ -7,9 +7,10 @@
...
         connector.recipes.post(org.managed_user_id, instance.recipe_id, None, 'stop')
```
Comment (Shwetabhk) [inline apps/bamboohr/signals.py:16]: Do we require to stop the recipe first to update emails?

File: apps/bamboohr/urls.py [hunk 5]
```diff
@@ -10,5 +10,5 @@
     path('bamboo_connection/', BambooHrConnection.as_view(), name='bamboo-connection'),
     path('configuration/', ConfigurationView.as_view(), name='configuration'),
     path('refresh_employees/', SyncEmployeesView.as_view(), name='sync-employees'),
-    path('start&stop/', StartAndStopRecipe.as_view(), name='start-and-stop')
+    path('update_recipe_status/', StartAndStopRecipe.as_view(), name='start-and-stop'),
```
Comment (Shwetabhk) [inline apps/bamboohr/urls.py:13]: just `update_recipe` is enough

File: apps/orgs/views.py [hunk 6]
```diff
@@ -218,3 +233,47 @@ def post(self, request, *args, **kwargs):
...
             )
+
+class WorkspaceAdminsView(generics.ListAPIView):
+
+    def get(self, request, *args, **kwargs):
+        """
+        Get Admins for the workspaces
+        """
+
+        org = Org.objects.get(pk=kwargs['org_id'])
```
Comment (Shwetabhk) [inline apps/orgs/views.py:244]: Please move all this logic to serializer

File: apps/bamboohr/signals.py (hunk 4 above)
Comment (NileshPant1999) [reply apps/bamboohr/signals.py:16]: yes

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**